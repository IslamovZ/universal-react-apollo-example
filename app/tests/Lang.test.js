import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import { print } from 'graphql-tag/printer';

import LangWithData, { LANG_QUERY, Lang } from 'components/Lang/Lang';

describe('Lang Container', () => {
  it('renders without crashing', () => {
    const output = renderer.create(
      <MockedProvider>
        <LangWithData />
      </MockedProvider>,
    );
    expect(output.toJSON()).toMatchSnapshot();
  });
});

describe('Lang Query', () => {
  it('should match expected shape', () => {
    expect(print(LANG_QUERY)).toMatchSnapshot();
  });
});

describe('Lang Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Lang terms={[{ children: [{ term_id: 1, slug: 'gb' }] }]} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

