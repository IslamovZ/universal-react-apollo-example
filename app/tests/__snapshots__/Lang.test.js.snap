exports[`Lang Component renders correctly 1`] = `
<div
  className="dropdown btn-group btn-group-default">
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-default"
    disabled={false}
    id="Lang"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    type="button">
    <span
      className="flag-icon flag-icon-undefined" />
     
    <span
      className="caret" />
  </button>
  <ul
    aria-labelledby="Lang"
    className="dropdown-menu"
    role="menu">
    <li
      className=""
      role="presentation"
      style={undefined}>
      <a
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex="-1">
        <span
          className="flag-icon flag-icon-gb" />
      </a>
    </li>
  </ul>
</div>
`;

exports[`Lang Container renders without crashing 1`] = `
<div>
  Loading....
</div>
`;

exports[`Lang Query should match expected shape 1`] = `
"query getTerms {
  wp_query {
    terms(slug: \"languages\") {
      term_id
      children {
        term_id
        slug
      }
    }
  }
}
"
`;
